instrumented-tests/http-server-mock/Sources/HTTPServerMock/ServerSession.swift

```diff
@@ -44,4 +44,32 @@ public class ServerSession {
                 )
             }
     }
+
+    /// Actively fetches 'POST` requests recorded by the server until a desired count is found, or timeouts returning current recorded requests
+    public func pullRecordedPOSTRequests(count: Int, timeout: TimeInterval) throws -> [POSTRequestDetails] {
+        var currentRequests = [ServerMock.RequestInfo]()
+
+        let timeoutTimer = DispatchSource.makeTimerSource(queue: DispatchQueue.global())
+        timeoutTimer.setEventHandler { timeoutTimer.cancel() }
+        timeoutTimer.schedule(deadline: .now() + timeout, leeway: .nanoseconds(0))
+        if #available(iOS 10.0, *) {
+            timeoutTimer.activate()
+        }
+
+        while !timeoutTimer.isCancelled && currentRequests.count < count {
+            Thread.sleep(forTimeInterval: 0.2)
+            currentRequests = try server
+                .getRecordedPOSTRequestsInfo()
+                .filter { requestInfo in requestInfo.path.contains(sessionIdentifier) }
+        }
+        timeoutTimer.cancel()
+
```

Comment:
How about throwing an exception from this method if we don't get what we expect:
```swift
throw ServerSessionException("Exceeded \(timeout)s timeout by receiving only \(currentRequests.count) requests, where \(count) were expected.")
```
This way, on the caller side (in integration tests), the execution will be stopped due to inconsistency. Otherwise, the execution will continue and we'll be accessing requests on indexes which do not exist, resulting with tests crash, instead of failure.

WDYT @nachoBonafonte ?

Reply:
Its a great idea, I will change to it